infoblox/resource_infoblox_cname_record.go

```diff
@@ -324,6 +401,16 @@ func resourceCNAMERecordImport(d *schema.ResourceData, m interface{}) ([]*schema
 	}
 
 	d.SetId(obj.Ref)
+	// Set ref in the state
+	if err = d.Set("ref", obj.Ref); err != nil {
+		return nil, err
+	}
```

Comment:
Please remove this as internally in resource Update the ref value will be updated according to the code.